### DISCUSSION TITLE
computeTileAreaAndDistance

### DISCUSSION THREAD

**Discussion by blueRaining** [heremaps/harp.gl#2266] [Q&A] [answered]
const size = tileBounds.getSize(tmpVectors3[1]);
        const objectSize = (0.5 * size.length()) / projectedPoint.w;

        return {
            area: objectSize * objectSize,
            //Dividing by w means we loose information for whether the point is behind the camera
            //(i.e. it is in front of the near plane) or beyond the far plane, hence we first clamp
            //to [-1, 1] range, before doing the division.
            distance:
                projectedPoint.z <= -projectedPoint.w
                    ? -1
                    : projectedPoint.z >= projectedPoint.w
                    ? 1
                    : projectedPoint.z / projectedPoint.w
        };

in computeTileAreaAndDistance i find that the objectSize  devided by  projectedPoint.w,the size of it is in ndc space? and the arae is real in screen space?

**Comment by nzjony** [comment, marked as answer]
It has been a while since I looked at this.

The objectSize value is in NDC space, so the area is a percentage of a cube 2 x 2, i.e. 4 units square, because the NDC space goes from -1,-1 to 1,1. (95% sure).

The distance is in NDC space, at least because it is clamped between -1 and 1 and divided by w.

**Comment by blueRaining** [comment]
but  tileBounds is in world space

**Reply by nzjony** [reply]
Yes, but it is divided by w, so it puts it into the NDC cube (if I'm not mistaken).

**Reply by blueRaining** [reply]
should we compute  tileBounds  to projection space?

**Reply by nzjony** [reply]
Why?

**Reply by blueRaining** [reply]
because the tileBounds is in world space, it is divided by w,w is computed by projection matrix,there are not in common space ,maybe i'm mistake,can you help me to understand it?

**Reply by nzjony** [reply]
I'm not following.

What do you mean they are not in common space? What is not in the common space?

This is a very rough estimate of the screen space. We take the bounds of the tile space, compute the diagonal length and then take this length and transform to NDC space, we do this to estimate roughly how much room the tile takes. Note, this doesn't even take into consideration the tilt, so don't get too caught up on the maths, it is anyway not accurate, but it is good enough to provide a rough idea of the size that the tile takes up.

I think the area is used to sort the tiles? So that tiles that are closer are processed first? But maybe we could remove this and just rely on the distance?

**Reply by blueRaining** [reply]
thank you for your replay `this is a very rough estimate of the screen space`, i get it
